src/analysis/mean_inequalities.lean

```diff
@@ -377,6 +377,50 @@ begin
     ((is_greatest_Lp s g hpq).2 ⟨φ, hφ, rfl⟩)
 end
 
+/-- Minkowski inequality: the `L_p` seminorm of the infinite sum of two vectors is less than or
+equal to the infinite sum of the `L_p`-seminorms of the summands, if these infinite sums both
+exist. A version for `nnreal`-valued functions. -/
+theorem Lp_add_le_tsum {f g : ι → ℝ≥0} {p : ℝ} (hp : 1 ≤ p) (hf : summable (λ i, (f i) ^ p))
+  (hg : summable (λ i, (g i) ^ p)) :
+  summable (λ i, (f i + g i) ^ p) ∧
+  (∑' i, (f i + g i) ^ p) ^ (1 / p) ≤ (∑' i, (f i) ^ p) ^ (1 / p) + (∑' i, (g i) ^ p) ^ (1 / p) :=
+begin
+  have pos : 0 < p := lt_of_lt_of_le zero_lt_one hp,
+  have H₁ : ∀ s : finset ι, ∑ i in s, (f i + g i) ^ p
+    ≤ ((∑' i, (f i)^p) ^ (1/p) + (∑' i, (g i)^p) ^ (1/p)) ^ p,
+  { intros s,
+    rw ← nnreal.rpow_one_div_le_iff pos,
+    refine le_trans (Lp_add_le s f g hp) (add_le_add _ _);
+    rw nnreal.rpow_le_rpow_iff (one_div_pos.mpr pos);
+    refine sum_le_tsum _ (λ _ _, zero_le _) _,
+    exacts [hf, hg] },
+  have bdd : bdd_above (set.range (λ s, ∑ i in s, (f i + g i) ^ p)),
+  { refine ⟨((∑' i, (f i)^p) ^ (1/p) + (∑' i, (g i)^p) ^ (1/p)) ^ p, _⟩,
+    rintros a ⟨s, rfl⟩,
+    exact H₁ s },
+  have H₂ : summable _ := (has_sum_of_is_lub _ (is_lub_csupr bdd)).summable,
+  refine ⟨H₂, _⟩,
+  rw nnreal.rpow_one_div_le_iff pos,
+  refine tsum_le_of_sum_le H₂ H₁,
+end
+
+/-- Minkowski inequality: the `L_p` seminorm of the infinite sum of two vectors is less than or
+equal to the infinite sum of the `L_p`-seminorms of the summands, if these infinite sums both
+exist. An alternative version for `nnreal`-valued functions, convenient if the infinite sums are
```

Comment:
Re "alternative version": outside the context of this file that doesn't make much sense. I think it's better to explicitly cross-ref both versions in their respective docstrings.